### PR TITLE
Adds glowing cybernetic eyes to R&D - No functions except for flashlight!

### PR DESCRIPTION
**Pull request by silicons**
Lol

Hey, you guys did say nonantag items that are fun.

Basically, gives you glowing eyes, because snowflake.
Purely cosmetic except for a 5-tile weak light beam blocked by anything that would normally block light.

### CODE REVIEW

File: code/modules/surgery/organs/eyes.dm
Comment (kevinz000) [inline code/modules/surgery/organs/eyes.dm]: note to eslf to remove

File: code/modules/surgery/organs/eyes.dm
Comment (kevinz000) [inline code/modules/surgery/organs/eyes.dm]: note to self to remove

**Comment by Repukan** [comment]
Powercreep

**Comment by kevinz000** [comment]
@repukan
REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE

**Comment by GunHog** [comment]
Instead of entering RGB numbers, why not use the same prompt that shows up when choosing your hair/eye color? This is a lot easier for players to understand and use.

**Comment by kevinz000** [comment]
hmmmmmmm

**Comment by evsey9** [comment]
@GunHog fuck casuals

**Comment by GunHog** [comment]
I assume that is sarcasm, but in the event that you are actually serious, I must ask; Do you feel that having a poor, unfriendly interface is acceptable when we have a superior option available? I do not understand what purpose that would serve. Of course, you may be joking, but my lack of social ability makes it difficult for me to tell.

**Comment by kevinz000** [comment]
@GunHog done, wew, i forgot byond had this.

**Comment by sawrge** [comment]
Already in the game

**Comment by kevinz000** [comment]
@sawrge
 Genetics laser eyes don't count.

**Comment by sawrge** [comment]
Flashlight implant exists already Kev

**Comment by kevinz000** [comment]
@sawrge Those take your sight away and doesn't have mob overlay effects, and is just a "huge light source centered on you". These project light further away.

**Comment by kevinz000** [comment]
rgb input changed to the preferences on aka it's user friendly and not shit.

**Comment by evsey9** [comment]
@GunHog nah, i was kidding. using the byond color thing for this would be a better idea as it is user friendly and still has the RGB option

**Comment by kevinz000** [comment]
@evsey9 I just found out the byond has both.

**Comment by Armhulen** [comment]
Can you make one in red?

**Comment by Repukan** [comment]
We already have flash light eyes.

Stop adding these "QoL" upgrades that just take away all trade-offs

**Comment by WJohn** [comment]
Yeah I thought you were gonna add this to the existing eyes, it'd only make sense.

**Comment by Repukan** [comment]
We already have night vision mesons
Fuck off reeeeeeeeeeeeeeeee

**Comment by Repukan** [comment]
And glasses

**Comment by kevinz000** [comment]
@WJohn that's the alternative
@Repukan this is magnitudes weaker than flashlight eyes.

**Comment by fludd12** [comment]
Flashlight eyes literally make you blind and then floodlight everywhere. These don't make you blind, but only project light in one direction, and significantly less of it at that. It's neither QoL nor an upgrade, it's more of a cybernetic of its own, which we need more of IMO.

**Comment by kevinz000** [comment]
The focus is on cosmetic if a 5 tile directional weak light beam is somehow too overpowered it can be nerfed or the light beam removed but really I don't feel it's an issue as this takes your cybernetic eye slot which is the same spot thermals x-ray and welding shield eyes go into. You're sacrificing that slot for a fun gimmick cosmetic item if you choose this so no this probably won't be a balance issue.

**Comment by kevinz000** [comment]
@Repukan now I see why people say you don't understand PRS if you think I'm doing this as a quote quality of life quote change.

**Comment by Militaires** [comment]
if i emp them the eyes should roll back into their head and shine the fuck out of their brains

**Comment by Militaires** [comment]
and go rainbow

**Comment by loqoman** [comment]
@kevinz000 @Repukan 

It's worth saying that, because they are cybernetic, you could still wear something in the eye slot.

What are the research level required to get the eyes?

**Comment by XDTM** [comment]
can you modify flashlight eyes to make them project an ultralong beam instead of an area?

**Comment by Iamgoofball** [comment]
@KorPhaeron we going to do something about R&D QoL powercreep "its balanced cuz u gotta click some ui buttons a few times" here or what

**Comment by Militaires** [comment]
also nigga this is better than any flashlight in the game except real flash eyes

**Comment by kevinz000** [comment]
>powercreep
Lol.

**Comment by kevinz000** [comment]
I fail to see how using this is powercreep when any other cybernetic eye is better .

**Comment by octareenroon91** [comment]
What, you want rainbow spotflashlights?

**Comment by BeeSting12** [comment]
We already have flashlight eyes, 👎

**Comment by lzimann** [comment]
I really dislike how this looks, with the minor update delay, plus colored lights, which is an automatic no for me.

File: code/modules/surgery/organs/eyes.dm
Comment (ChangelingRain) [inline code/modules/surgery/organs/eyes.dm]: if(scanning.opacity || scanning.has_opaque_atom) and then you can remove the loop

**Comment by Bawhoppen** [comment]
adding more R&D items is not what kor meant

**Comment by kevinz000** [comment]
@lzimann what do you mean by minor update delay? It updates instantly if the mob moves but there's no hook for mob turning so it has to process to do that.
@Bawhoppen better than combat items!

**Comment by RemieRichards** [comment]
@kevinz000 
>but there's no hook for mob turning

Factually incorrect.

File: code/modules/surgery/organs/eyes.dm
Comment (RemieRichards) [inline code/modules/surgery/organs/eyes.dm]: Hahaha, No.

File: code/modules/surgery/organs/eyes.dm
Comment (RemieRichards) [inline code/modules/surgery/organs/eyes.dm]: this proc is pointless, just use the ONE line in this proc where you use the proc.
Don't add overhead for nothing.

File: code/modules/surgery/organs/eyes.dm
Comment (kevinz000) [inline code/modules/surgery/organs/eyes.dm]: :'(

**Comment by RemieRichards** [comment]
@kevinz000 use ```setDir()``` you goat.

**Comment by AnturK** [comment]
Move hook -> Moved hook

File: code/modules/mob/living/carbon/carbon_movement.dm
Comment (Cyberboss) [inline code/modules/mob/living/carbon/carbon_movement.dm]: Should be in Moved

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: YOU KNOW HOW QDEL WORKS DAMMIT KEV

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: return

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: STOP

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: Ternerys that don't return a value are a nono from me, use an if

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: !QDELETED(owner)

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: New vars after parent vars

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: Just leave this as null, 0 isn't a valid dir

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: !QDELETED(OWNER)

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: !QDELETED(OWNER)

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: QDEL_NULL

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: QDEL_LIST does a Cut() already

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: You can do all of these 3 lines with set_light

`/atom/proc/set_light(var/l_range, var/l_power, var/l_color = NONSENSICAL_VALUE)`

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: Again

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: !QDELETED(on_mob)

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: You didn't qdel this first

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: !QDELETED(on_mob)

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: You never use scanning before you reassign it

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: !QDELETED(owner)

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: This list can be lazy

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: LAZYADD

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: Nah, remove this check

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: The inputs in the proc need to do more checks, what if the owner drops them and moves away? What if the were qdel'd? etc...

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/objs.dm
Comment (Cyberboss) [inline code/game/objects/objs.dm]: Add comments to these to note that they only work on internal organs

File: code/modules/mob/mob_movement.dm
Comment (Cyberboss) [inline code/modules/mob/mob_movement.dm]: You're calling this twice, once in /mob and again int /carbon

File: code/modules/surgery/organs/eyes.dm
Comment (Cyberboss) [inline code/modules/surgery/organs/eyes.dm]: ,

File: code/modules/mob/mob_movement.dm
Comment (Cyberboss) [inline code/modules/mob/mob_movement.dm]: Smae here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/eyes.dm
Comment (octareenroon91) [inline code/modules/surgery/organs/eyes.dm]: >,

File: code/modules/mob/mob_movement.dm
Comment (kevinz000) [inline code/modules/mob/mob_movement.dm]: the carbon one only works on internal organs as internal organs aren't defined in mob

File: code/game/objects/objs.dm
Comment (kevinz000) [inline code/game/objects/objs.dm]: they work on everything

File: code/modules/mob/mob_movement.dm
Comment (kevinz000) [inline code/modules/mob/mob_movement.dm]: above

**Comment by dannno** [comment]
someone close this we already have flashlight eyes

**Comment by kevinz000** [comment]
@dannno I see you didn't read what I said to the two others who said that.

**Comment by dannno** [comment]
I don't care what you said because we already have flashlight eyes

**Comment by kevinz000** [comment]
@dannno good for you

**Comment by octareenroon91** [comment]
>Have one kind of flashlight eyes, which happen to be a joke item, nearly useless for their owner.
>Oppose adding one, very different, actually useful variation.

>Like there aren't nearly identical guns or poisons, and explosions differ much aside from size.

**Comment by Iamgoofball** [comment]
>add flashlight eyes as a joke item, make them blind you, because having a free never-remove flashlight would be hillariously OP

>people try to QoL an R&D version

**Comment by octareenroon91** [comment]
>QoL

>Forgetting the historical use of the chest cavity for flashlights
>>when it's not the disk of course.

**Comment by Armhulen** [comment]
>me me arrows

**Comment by kevinz000** [comment]
@Cyberboss it broke without the scanning var being assigned first as it'll be null because it paths from scanning to the direction the user's in.

File: code/modules/surgery/organs/eyes.dm
Comment (RemieRichards) [inline code/modules/surgery/organs/eyes.dm]: 0 is a valid dir of sorts, atleast, it has a useful function in ```get_step()``` (it's what we use for ```get_turf()```, ```#define get_turf(A) (get_step(A, 0))```)

**Comment by lzimann** [comment]
@tgstation/commit-access Feel free to reopen and merge this if anyone likes, otherwise it is just going to sit in the tracker forever since the fruit is no longer with us.

**Comment by kevinz000** [comment]
👍

**Comment by KorPhaeron** [comment]
It's harmless gimmick that has nothing to do with flashlight eyes and isn't even made by the same department or with remotely similar resources and it the number of arbitrary positive emojis is higher than the negative ones

I can't be bothered to review this though so if nobody else does it will probably die again

**Comment by KorPhaeron** [comment]
Might be cooler as a borg headlamp thing though or headlights on the vehicles

**Comment by optimumtact** [comment]
Why even reopen it then, what a joke

**Comment by GunHog** [comment]
@Cyberboss can review the code, and I could do testing tomorrow if you want? Neither of us are maintainer, so you will have to merge it, Kor.

**Comment by kevinz000** [comment]
@Cyberboss already did i responded to his stuff lol.

**Comment by lzimann** [comment]
Ok, I said anyone can reopen and merge this. But what is even the point of reopening if you are not doing it yourself? It will probably just die again.

**Comment by KorPhaeron** [comment]
Remie was reviewing it

**Comment by octareenroon91** [comment]
This is eagerly anticipated. P
lease finish.

**Comment by GunHog** [comment]
Kev. You fought hard for this. Kev. Finish your code. Kev. Do not let this go stale and get closed. Kev, wake up!

**Comment by Cyberboss** [comment]
You are filled with determination

**Comment by dannno** [comment]
kor is a fucking race traitor

**Comment by kevinz000** [comment]
SHIT WOOPS

**Comment by kevinz000** [comment]
fixed guys

**Comment by optimumtact** [comment]
closed for stale lol

**Comment by Nilons** [comment]
this is a cool pr, i like the way the light works

**Comment by kevinz000** [comment]
well everything's updated now @AnturK

**Comment by optimumtact** [comment]
closed for stale lol

**Comment by kevinz000** [comment]
=|

**Pull request merged**
